### ISSUE TITLE
Faied to create mysql db server

### ISSUE THREAD

**Issue by RainFlying** [terraform-providers/terraform-provider-azurerm#751]
Check the following for the error message:
```
azurerm_mysql_server.db_server: Creating...
  administrator_login:          "" => "kosun"
  administrator_login_password: "<sensitive>" => "<sensitive>"
  fqdn:                         "" => "<computed>"
  location:                     "" => "eastasia"
  name:                         "" => "test-db-server"
  resource_group_name:          "" => "test-group"
  sku.#:                        "" => "1"
  sku.3479340777.capacity:      "" => "100"
  sku.3479340777.name:          "" => "MYSQLS400"
  sku.3479340777.tier:          "" => "Standard"
  ssl_enforcement:              "" => "Enabled"
  storage_mb:                   "" => "128000"
  tags.%:                       "" => "<computed>"
  version:                      "" => "5.7"
azurerm_mysql_server.db_server: Still creating... (10s elapsed)
azurerm_mysql_server.db_server: Still creating... (20s elapsed)
azurerm_mysql_server.db_server: Still creating... (30s elapsed)
azurerm_mysql_server.db_server: Still creating... (40s elapsed)
azurerm_mysql_server.db_server: Still creating... (50s elapsed)
azurerm_mysql_server.db_server: Still creating... (1m0s elapsed)

Error: Error applying plan:

1 error(s) occurred:

* azurerm_mysql_server.db_server: 1 error(s) occurred:

* azurerm_mysql_server.db_server: mysql.ServersClient#CreateOrUpdate: Failure sending request: StatusCode=200 -- Original Error: Long running operation terminated with status 'Failed': Code="ServerNameAlreadyExists" Message="Specified server name is already used."

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.

```

I checked the web console, thare is no mysql servers named 'test-db-server'
<img width="1526" alt="screenshot 2018-01-25 11 48 56" src="https://user-images.githubusercontent.com/507678/35371279-c2205aa8-01c5-11e8-8dbc-a5e0d7038fa4.png">

**Comment by jansepke** [comment]
the mysql server name needs to be globally unique because a dns name will be generated for it. see: https://docs.microsoft.com/en-us/azure/mysql/quickstart-create-mysql-server-database-using-azure-portal

the terraform docs should document this

**Comment by RainFlying** [comment]
@jansepke OK. Thanks.

**Comment by tombuildsstuff** [comment]
Fixed via #752
